parsers/json_parser_listener.go

```diff
@@ -213,47 +213,63 @@ func (l *JsonParserListener) EnterNumber(ctx *parser_json.NumberContext) {
 		}
 	}
 
-	// Create holder to store the node where the location information
-	// of the number value should be stored
-	locationInfoDest := l.getTOS()
-	if l.configFile == nil { // This number is the top level entity
-		// Create new node for number
+	if l.configFile == nil && l.stack.Len() == 0 { // The only value in the config file
+		// Create new node for pair. The type will be set in Enter<Value>
 		node := &Node{Type: numberType, Value: value}
 
-		// Set config file to point to this number
+		// Set config file to point to this string
 		l.configFile = node
 
-		// Set location destination to the newly created node
-		locationInfoDest = node
+		// Push object node to stack
+		l.stack.Push(node)
 
-	} else if l.getTOS().Type == Null { // Is the value of a pair
-		// Set pair node to correct type
-		l.getTOS().Type = numberType
+		// Add location information of the pair key
+		node.NameLocation.Start.Line = ctx.GetStart().GetLine()
+		node.NameLocation.Start.Column = ctx.GetStart().GetColumn() + 1 // ANTLR count columns from 0 instead of 1
 
-		// Create value for number
-		l.getTOS().Value = value
+	} else {
+		// Create holder to store the node where the location information
+		// of the number value should be stored
+		locationInfoDest := l.getTOS()
+		if l.configFile == nil { // This number is the top level entity
```

Comment:
This is never invoked cause if we got to the else, then configFile is not nil. It is preferable to have the original 3 way if else and add the location information in each if/else body. Same for all the other functions with this change.